template/components/ActionNetworkForm.vue

```diff
@@ -0,0 +1,138 @@
+        Enter your information below to sign the petition.
+      </p>
+    </div> <!-- .push -->
+    <form @submit.prevent="submitForm()" v-if="!hasSigned"
+          class="sml-push-y2 med-push-y3">
+      <p class="text-warn" v-if="errorMessage">{{ errorMessage }}</p>
+      <div class="flex-row">
+        <input v-model="name" type="text" placeholder="Name*" required>
+        <input v-model="email" type="email" placeholder="Email*" required>
+      </div> <!-- .flex-row -->
+      <div class="flex-row sml-push-y2">
+        <input v-model="address" type="text" placeholder="Address"
+               class="sml-flex-4">
+        <input v-model="zipCode" type="tel" placeholder="ZIP Code">
+      </div> <!-- .flex-row -->
+
+      <button class="btn btn-block sml-push-y2 med-push-y3" :disabled="isSending">
+        <span v-if="isSending">
+          Sending...
+        </span>
+        <span v-else>
+          Take action
+        </span>
+      </button>
+      <p class="sml-push-y1"><small>
+        <a href="https://www.fightforthefuture.org/" target="_blank">
+          Fight for the Future</a>
+        will email you updates, and you can unsubscribe at any time. If
+        you enter your number (it&rsquo;s optional) we will follow up by SMS.
+        Message &amp; data rates apply. You can always text STOP to stop
+        receiving messages.
+        <a href="https://www.battleforthenet.com/privacy/" target="_blank">
+          Privacy Policy</a>
+      </small></p>
+    </form>
+  </div>
+</template>
+
+<script>
+import axios from 'axios'
+import config from '~/config'
+import { sendToMothership } from '~/assets/js/helpers'
+import ShareButton from '~/components/ShareButton'
+
+export default {
+  components: {
+    ShareButton
+  },
+
+  data() {
+    return {
+      isSending: false,
+      hasSigned: false,
+      errorMessage: null,
+      // form fields
+      name: null,
+      email: null,
+      address: null,
+      zipCode: null
+    }
+  },
+
+  computed: {
+    donateUrl () { return config.donateUrl }
+  },
+
+  methods: {
+    async submitForm() {
+      if (this.isSending) return
+
+      this.isSending = true
+
+      try {
+        const response = await sendToMothership({
+          subject: "TODO: subject",
+          member: {
+            first_name: this.name,
+            email: this.email,
+            street_address: this.address,
+            postcode: this.zipCode,
+            country: 'US'
+          },
+          hp_enabled: 'true',
+          guard: '',
+          // contact_congress: 1, // TODO: Optional
+          // fcc_ecfs_docket: "17-108", // TODO: Optional
+          an_tags: "[\"net-neutrality\"]",
+          an_petition_id: config.actionNetworkPetitonId
+          // action_comment: TODO add if desired
+        })
+
+        this.$trackEvent('petition_form', 'submit')
+        this.isSending = false
+        this.hasSigned = true
+      }
+      catch (err) {
+        self.isSending = false
```

Comment:
Needs to be tested. BFTN had `self` here with no `this = self` above, but I'm still not certain it will work as intended. I was unable to produce an error even submitting the form blank, but if there are values that will return an error, let me know and I'll be happy to test this further.

Reply:
I don't see `self` defined anywhere. Probably a bad copy/paste in the BFTN project.